### PR TITLE
BH-423: use logger in commands

### PR DESCRIPTION
**Pull request by greg-akeneo**
**Description (for Contributor and Core Developer)**

This PR impact two differents parts

* Use logger in akeneo:batch:job-queue-consumer-daemon
* Re-work monolog configuration (thanks @tseho for the huge contribution), bye bye finger crossed
* Introduce new env var LOGGING_LEVEL

## logger in akeneo:batch:job-queue-consumer-daemon

To improve log management in Akeneo PIM we must use monolog not only in the request part, but also in console command.
We are starting to change the akeneo:batch:job-queue-consumer-daemon to use monolog.

After many differents tries to compute the job process log, we decide to use akeneo:batch:job-queue-consumer-daemon as a passthrough to keep job log as is. A job log is already compute by monolog, we don't need to recompute them.

## Monolog configuration

Thanks @tseho for his PR #14021, I used it to reworked monolog configuration

Since it's not easy to read,
Here is a resume of the proposed CE configuration:

|  | main | console | business_event |
|---|---|---|---|
| prod | output to `prod.log`, stream (trigger on NOTICE), exclude `event_api`, `business_event` | exclude `event`, `doctrine`, `event_api`, `business_event` | :no_entry_sign: |
| dev | output to `dev.log`, stream (trigger on NOTICE)*, exclude `event`, `event_api`, `business_event` | exclude `event`, `doctrine`, `event_api`, `business_event` | :no_entry_sign: |
| behat | output to `behat.log`, stream (trigger on DEBUG), exclude `event`, `event_api`, `business_event` | exclude `event`, `doctrine`, `event_api`, `business_event` | :no_entry_sign: | :no_entry_sign: |
| test | :no_entry_sign: | exclude `event`, `doctrine`, `event_api`, `business_event` | :no_entry_sign: |
| test_fake | :no_entry_sign: | :no_entry_sign: | :no_entry_sign: |

* the env.dev doesn't exist, and we don't want to introduce it,  to not overwrite your dev env, we choose to let you manage this file and this value.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php [hunk 1]
```diff
@@ -87,64 +100,74 @@ protected function configure()
...
+                            usleep(self::RUNNING_PROCESS_CHECK_INTERVAL);
 
-                        continue;
-                    }
+                            continue;
+                        }
 
-                    $output->write($process->getIncrementalOutput());
-                    $errOutput->write($process->getIncrementalErrorOutput());
+                        $this->logger->debug($process->getIncrementalOutput());
```
Comment (mr-prud) [inline src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php:150]: The job output log stream is already in json format. We need to forward int the error output.

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php [hunk 2]
```diff
@@ -87,64 +100,74 @@ protected function configure()
...
-                }
+                    // update status if the job execution failed due to an uncatchable error as a fatal error
+                    $exitStatus = $this->executionManager->getExitStatus($jobExecutionMessage);
+                    if ($exitStatus->isRunning()) {
+                        $this->executionManager->markAsFailed($jobExecutionMessage);
+                    }
 
-                $output->write($process->getIncrementalOutput());
-                $errOutput->write($process->getIncrementalErrorOutput());
+                    $this->logger->debug($process->getIncrementalOutput());
```
Comment (mr-prud) [inline src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php:163]: same here

**Review by mr-prud — APPROVED** [review]

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Resources/config/services.yml [hunk 3]
```diff
@@ -78,12 +78,14 @@ services:
...
     Akeneo\Tool\Bundle\BatchQueueBundle\Command\JobQueueConsumerCommand:
         arguments:
+            - '@logger'
             - '@akeneo_batch_queue.queue.database_job_execution_queue'
             - '@akeneo_batch_queue.queue.job_execution_message_repository'
             - '@akeneo_batch_queue.manager.job_execution_manager'
             - '%kernel.project_dir%'
         tags:
             - { name: console.command }
+            - { name: monolog.logger, channel: Queue }
```
Comment (nmarniesse) [inline src/Akeneo/Tool/Bundle/BatchQueueBundle/Resources/config/services.yml:88]: `queue` is a little too abstract IMO. Could it be `job_queue`? Maybe personal preference, feel free to ignore

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php [hunk 4]
```diff
@@ -62,14 +63,19 @@ class JobQueueConsumerCommand extends Command
     /** @var string */
     private $projectDir;
 
+    /** @var LoggerInterface */
+    private $logger;
```
Comment (nmarniesse) [inline src/Akeneo/Tool/Bundle/BatchQueueBundle/Command/JobQueueConsumerCommand.php:67]:
```suggestion
    private LoggerInterface $logger;
```

**Review by nmarniesse — APPROVED** [review]

File: src/Akeneo/Tool/Bundle/BatchQueueBundle/Resources/config/services.yml (hunk 3 above)
Comment (mr-prud) [reply src/Akeneo/Tool/Bundle/BatchQueueBundle/Resources/config/services.yml]: you're right, your suggestion is better

**Pull request merged**